### PR TITLE
[Api] [Vue] [SIMS #682] Hide Save-Draft and Submit button while viewing the application

### PR DESCRIPTION
**Pull request by dheepak-aot**
Form.io wrapper component readOnly attribute type changed to boolean

isReadOnly variable added which decides to show the form in read-only mode or not

API check for status added for submit application

API check is not required to be added for save draft as it is already in place

### CODE REVIEW

File: sources/packages/api/src/route-controllers/application/application.controller.ts
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/application/application.controller.ts:118]: The method below "submitApplication" is already ensuring that the application will be in the draft state and throwing an exception.
I would recommend:
1 - From the service layer, change the current exception being thown 
from
```
if (!application) {
  throw new Error(
    "Student Application not found or it is not in the correct status to be submitted.",
  );
}
```

to

```
throw new CustomNamedError(
    "Student Application not found or it is not in the correct status to be submitted.",
    APPLICATION_NOT_FOUND,
  );
```

2 - Capture the exception in the controller and throw a NotFoundException.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:11]: I believe that there is one AC missing. If the _submittingApplication_ is true, the draft button should not be visible, as per below AC
```
Do not display "save draft" while Application Submission is in Progress
```

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:117]: I would recommend starting with true. It seems to be safer in the sense that the user will not be able to interact with the form till the data is loaded.

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:129]: I believe that the below expression must have the second part converted to a boolean (using double exclamations), as below.
```
isReadOnly.value =
        applicationData.applicationStatus !== ApplicationStatus.draft ||
        !!props.readOnly;
```

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work with the Form.IO read-only.
Please take a look at the comments 😉

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue
Comment (dheepak-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:129]: I do not see a benefit of converting an object to boolean here as we take the advantage of objects being used in expression. The logic works perfectly with given verbose.

Comment (andrewsignori-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:129]: What type of object do you expect as a result in "isReadOnly.value"?

Comment (andrewsignori-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:129]: - This PR changed the Form.IO component readOnly property from string to boolean (I am assuming that was because it reflects better what it really is).
- isReadOnly.value if initialized as boolean, because I assume that it is better expressed as a boolean
- The mentioned expression could return a boolean or a string, so I was just assuming that it would be more accurate to keep the isReadOnly.value always populated with a boolean instead of sometimes as a string, as shown below.

image

image

**Review by ann-aot — COMMENTED** [review]
LGTM 👍 pls address the other comments

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue
Comment (dheepak-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue:129]: I got it now. It is getting tricky with the scenario. Will modify accordingly

**Review by guru-aot — APPROVED** [review]
LGTM

**Review by abschwenker — APPROVED** [review]
LGTM, once comments addressed please merge

File: sources/packages/api/src/route-controllers/application/application.controller.ts
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/application/application.controller.ts]: The error handling part is done as discussed.

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for the adjustments. Great work in general 👍

File: sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue
Comment (dheepak-aot) [reply sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue]: Flag added to hide "Save Draft" while submission in progress

**Review by ann-aot — APPROVED** [review]

**Pull request merged**
